nomad/structs/structs.go

```diff
@@ -10586,6 +10587,10 @@ type Evaluation struct {
 	// to constraints or lacking resources.
 	BlockedEval string
 
+	// RelatedEvals is a list of all the evaluations that are related (next,
+	// previous, or blocked) to this one. It may be nil if not requested.
+	RelatedEvals []*EvaluationStub
```

Comment:
Originally this was a `map`, but a list would be a little easier for the UI to consume.

Reply:
I ran into a similar kerfuffle when working on fuzzy search, we should probably have some kind of methodology for agreeing on a schema before starting implementation work

Reply:
I think the biggest problem for me is that I don't fully understand the UI data layer 😅 

I wrote some [questionable JS code](https://github.com/hashicorp/nomad/commit/8116ebc48c7ab8059b3de4808e49759a1caaa3d6) to validate this endpoint and a list seemed to align better with how data is loaded into the UI. But I will validate this with someone that knows what they are doing.